src/count-blog-domain.js

```diff
@@ -0,0 +1,61 @@
+/* This file contains the code for counting the blog posts from different domains.
+ * The module receives an array of blog urls and returns an array of json object in format
+ * {name: domainName, count: number of frequency}
+*/
+
+//  import url library
+const url = require('url');
+
+/*  getDomain() is used to get the domain from hostname without username.
+*   It will split the hostname that pass in, if there is only two parts,
+*   then it will return the hostname;
+*   if not, it will remove the first part.
+*/
+function getDomain(hostname) {
+  const hostNameCounter = hostname.split('.').length;
+  const domain = hostname.substring(hostname.indexOf('.') + 1);
+  if (hostNameCounter === 2) {
+    return hostname;
+  }
+  return domain;
+}
+
+/* createDomainObject() will create an object if it's not exist */
+function createDomainObject(domainName) {
+  return { name: domainName, count: 1 };
+}
+
+/* it will return object array: {name: domainName, count: numberOfDomain} */
+module.exports.blogDomainCounter = function (feedUrls) {
+  const domainSummary = [];
+  /* When the domainSummary[] is empty,the first object must be pushed into the array,
+  * it doesn''t need to check is it exist or not.
+  */
+
+
+  const { hostname } = url.parse(feedUrls[0].url);
```

Comment:
What if `feedUrls` is an empty array?  I think you need a check above to make sure that there is at least 1 element in it, or this will also crash.

Reply:
But we won't  pass an empty array to the function, right? If there is no feedUrls, we don't need to count it.

Reply:
Again, you assume things.  That's not how computers work though: they do what we tell them; and if the calling code has a bug, you end up crashing here.  In general, anything that can go wrong, will eventually go wrong.  Write your code assuming the worst.